src/lock.js

```diff
@@ -36,6 +36,5 @@ exports.lock = (dir, callback) => {
           .then(() => cb())
           .catch(err => cb(err))
       }})
-    })
-    .catch(err => callback(err))
+    }, callback)
```

Comment:
@hugomrdias let's add the catch and log the error for now to avoid the unhandled rejection.  We can improve this when we move async. This also needs to be rebased with master.